File: bench/iteration-recursion/3.go
package main

import (
	"os"
	"strconv"
	"fmt"
	"time"
)

func main() {
	var n = 40
	var args = os.Args[1:]
	if len(args) >= 1 {
		n, _ = strconv.Atoi(args[0])
	}

	var start = time.Now()

	var x, y = 0, 1
	for i := 0; i < n; i++ {
		z := x + y
		x = y
		y = z
	}

	println(x)

	fmt.Println(time.Now().Sub(start).Nanoseconds())
}

File: bench/iteration-recursion/3.cpp
#include <iostream>
#include <chrono>

using namespace std;

int main(int argc, char const *argv[])
{
	int n = 40;
	if (argc == 2)
	{
		n = atoi(argv[1]);
	}

	auto start = chrono::steady_clock::now();

	int x = 0, y = 1, z = 0;
	for (int i = 0; i < n; i++) {
		z = x + y;
		x = y;
		y = z;
	}

	cout << x << endl;

	cout << chrono::duration_cast<chrono::nanoseconds>(chrono::steady_clock::now() - start).count() << endl;

	return 0;
}
